crates/vsock/src/main.rs

```diff
@@ -287,116 +287,182 @@ mod tests {
     }
 
     #[test]
-    #[serial]
     fn test_vsock_config_setup() {
-        let args = VsockArgs::from_args(3, "/tmp/vhost4.socket", "/tmp/vm4.vsock", 64 * 1024);
+        let test_dir = tempdir().expect("Could not create a temp test directory.");
+        std::env::set_current_dir(&test_dir).expect("Could not chdir to temp test directory.");
```

Comment:
Won't setting the cwd still race between the threads? Why is it needed now while it was not needed before?

Reply:
the dir returned by tempdir is unique, not shared between threads

Reply:
Well but `std::env::set_current_dir` will set the working directory for all threads at once, right? Also, it is not clear to me why `std::env::set_current_dir` was not required before but is now.

Reply:
Good point on the threads. I will remove it.